src/main/java/org/jabref/JabRefExecutorService.java

```diff
@@ -135,18 +134,36 @@ public void submit(TimerTask timerTask, long millisecondsDelay) {
 
     public void shutdownEverything() {
         // those threads will be allowed to finish
-        this.executorService.shutdown();
-        //those threads will be interrupted in their current task
+        gracefulShutdown(this.executorService);
+        // those threads will be interrupted in their current task
         this.lowPriorityExecutorService.shutdownNow();
         // kill the remote thread
         stopRemoteThread();
         // timer doesn't need to be canceled as it is run in daemon mode, which ensures that it is stopped if the application is shut down
     }
 
-    private class NamedRunnable implements Runnable {
+    private void gracefulShutdown(ExecutorService pool) {
+        // Disable new tasks from being submitted
+        pool.shutdown();
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!pool.awaitTermination(60, TimeUnit.SECONDS)) {
+                // Cancel currently executing tasks
+                pool.shutdownNow();
```

Comment:
I don't think we should cancel regular tasks during the shutdown but let them complete. There may be cases where we definitely want to wait until a task has completed no matter how long it takes. Consider the save operation. Maybe a save of a database of 20 000 entries on a network drive takes very long. If we cancel that, JabRef would stop during the save and corrupt the data.

I think we should not use `shutdownNow` here (and below), but we could print some log method to show that JabRef is trying to shutdown, but some tasks are not complete yet.